hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ResponseUtil.java

```diff
@@ -84,6 +86,25 @@ public static Response badRequest(String message) {
                 .entity(new ApiError(message)).type(APPLICATION_JSON_TYPE).build();
     }
 
+    public static Response onException(Exception e, Logger log) {
+        if (e instanceof NotFoundException || (null != e.getCause() && e.getCause() instanceof NotFoundException)) {
```

Comment:
Perhaps the expression can be simplified as 2nd part is also covered in next validation.